app/controllers/dashboards_controller.rb

```diff
@@ -1,4 +1,30 @@
 # frozen_string_literal: true
 class DashboardsController < ApplicationController
-  def index; end
+  include Flashable
+
+  def index
+    @uploads = Upload.last_uploads
+  end
+
+  def build
+    @version = InstitutionBuilder.run(current_user)
```

Comment:
This operation is going to take a long time to run isn't it? This action is most likely going to timeout well before this action returns. I think you're going to need to run it in a thread and just return a status indicating that it is running, at least until such time we can setup a worker with id, and a polling mechanism of some sort to see when the job is complete.

@patrickvinograd might need to fine tune the timeout for this seeing as this operation could take a long time.

Also I don't see any rescue blocks or anything, what would result if this encounters an error?